doc/source/howtos/proxysql-v1.rst

```diff
@@ -0,0 +1,424 @@
+.. orphan::
+
+Using ProxySQL v1 with ``proxysql-admin``
+================================================================================
+
+ProxySQL Version 1.4 does not natively support |PXC| and |proxysql-admin| tool
+requires custom bash scripts to keep track of PXC status:
+``proxysql_galera_checker`` and ``proxysql_node_monitor``.
+
+Installing ProxySQL v1
+================================================================================
+
+If that is what you used to :ref:`install PXC <install>` or any other Percona
+software, run the corresponding command:
+
+* On Debian or Ubuntu:
+
+.. code-block:: bash
+
+   $ sudo apt-get install proxysql
+
+* On Red Hat Enterprise Linux or CentOS:
+
+.. code-block:: bash
+
+   $ sudo yum install proxysql
+
+Alternatively, you can download packages from
+https://www.percona.com/downloads/proxysql/.
+
+To start ProxySQL, run the following command:
+
+.. code-block:: bash
+
+   $ sudo service proxysql start
+
+.. _default-credentials:
+
+.. warning:: Do not run ProxySQL with default credentials in production.
+
+   Before starting the ``proxysql`` service,
+   you can change the defaults in the :file:`/etc/proxysql.cnf` file
+   by changing the ``admin_credentials`` variable.
+   For more information, see `Global Variables
+   <https://github.com/sysown/proxysql/blob/master/doc/global_variables.md>`_.
+
+Automatic Configuration
+=======================
+
+The ``proxysql`` package from Percona includes the ``proxysql-admin`` tool
+for configuring |PXC| nodes with ProxySQL.
+
+.. note:: The *ProxySQL Admin* is specially developed by Percona to
+   automatize this configuration. Bug reports and feature proposals are welcome
+   in the ProxySQL Admin `issue tracking system <https://jira.percona.com/projects/PSQLADM>`_.
+
+.. note:: The ``proxysql-admin`` tool can only be used
+   for *initial* ProxySQL configuration.
+
+To view usage information, run ``proxysql-admin`` without any options:
+
+.. code-block:: text
+
```

Comment:
This help text is from an older version. Here is the current help text:
Usage: proxysql-admin [ options ]
Options:
  --config-file=<config-file>        Read login credentials from a configuration file
                                     (command line options override any configuration file login credentials)
  --proxysql-datadir=<datadir>       Specify the proxysql data directory location
  --proxysql-username=user_name      ProxySQL service username
  --proxysql-password[=password]     ProxySQL service password
  --proxysql-port=port_num           ProxySQL service port number
  --proxysql-hostname=host_name      ProxySQL service hostname
  --cluster-username=user_name       Percona XtraDB Cluster node username
  --cluster-password[=password]      Percona XtraDB Cluster node password
  --cluster-port=port_num            Percona XtraDB Cluster node port number
  --cluster-hostname=host_name       Percona XtraDB Cluster node hostname
  --cluster-app-username=user_name   Percona XtraDB Cluster node application username
  --cluster-app-password[=password]  Percona XtraDB Cluster node application passwrod
  --without-cluster-app-user         Configure Percona XtraDB Cluster without application user
  --monitor-username=user_name       Username for monitoring Percona XtraDB Cluster nodes through ProxySQL
  --monitor-password[=password]      Password for monitoring Percona XtraDB Cluster nodes through ProxySQL
  --use-existing-monitor-password    Do not prompt for a new monitor password if one is provided.
  --node-check-interval=3000         Interval for monitoring node checker script (in milliseconds)
                                     (default: 3000)
  --mode=[loadbal|singlewrite]       ProxySQL read/write configuration mode
                                     currently supporting: 'loadbal' and 'singlewrite'
                                     (default: 'singlewrite')
  --write-node=host_name:port        Writer node to accept write statments.
                                     This option is supported only when using --mode=singlewrite
                                     Can accept comma delimited list with the first listed being
                                     the highest priority.
  --include-slaves=host_name:port    Add specified slave node(s) to ProxySQL, these nodes will go
                                     into the reader hostgroup and will only be put into
                                     the writer hostgroup if all cluster nodes are down (this
                                     depends on the value of --use-slave-as-writer).
                                     Slaves must be read only.  Can accept a comma delimited list.
                                     If this is used make sure 'read_only=1' is in the slave's my.cnf
  --use-slave-as-writer=<yes/no>     If this value is 'yes', then a slave may be used as a writer
                                     if the entire cluster is down. If 'no', then a slave
                                     will not be used as a writer. This option is required
                                     if '--include-slaves' is used.
  --writer-is-reader=<value>         Defines if the writer node also accepts writes.
                                     Possible values are 'always', 'never', and 'ondemand'.
                                     'ondemand' means that the writer node only accepts reads
                                     if there are no other readers.
                                     (default: 'ondemand')
  --max-connections=<NUMBER>         Value for max_connections in the mysql_servers table.
                                     This is the maximum number of connections that
                                     ProxySQL will open to the backend servers.
                                     (default: 1000)
  --debug                            Enables additional debug logging.
  --help                             Dispalys this help text.

These options are the possible operations for proxysql-admin.
One of the options below must be provided.
  --adduser                          Adds the Percona XtraDB Cluster application user to the ProxySQL database
  --disable, -d                      Remove any Percona XtraDB Cluster configurations from ProxySQL
  --enable, -e                       Auto-configure Percona XtraDB Cluster nodes into ProxySQL
  --quick-demo                       Setup a quick demo with no authentication
  --syncusers                        Sync user accounts currently configured in MySQL to ProxySQL
                                     May be used with --enable.
                                     (deletes ProxySQL users not in MySQL)
  --sync-multi-cluster-users         Sync user accounts currently configured in MySQL to ProxySQL
                                     May be used with --enable.
                                     (doesn't delete ProxySQL users not in MySQL)
  --version, -v                      Print version info